OpenSim/Simulation/Control/TrackingController.cpp

```diff
@@ -76,7 +77,9 @@ void TrackingController::setNull()
 //=============================================================================
 void TrackingController::setDesiredStatesStorage(const Storage* aYDesStore)
 {
-    _desiredStatesStorage = aYDesStore;
+    auto* old = _desiredStatesStorage.release();
+    delete old;
```

Comment:
> Despite Tom's excitement, I don't think this is what we should do.

I didn't perform a thorough analysis of this change. I checked [the commit where this change was made](https://github.com/opensim-org/opensim-core/pull/1205/commits/02b4c1cc8509d758a245e65acd6d63c0251dfa79), and it seemed as though @aseth1 just happened to find the leak and patched it up without redoing the whole kitchen. If further analysis is necessary, I suggest opening a new issue to investigate since this class seems unrelated to the point of this PR.